src/Corax/Indexing/IndexWriter.cs

```diff
@@ -155,6 +156,20 @@ private void InitializeFieldRootPage(IndexedField field)
                 field.FieldRootPage = _fieldsCache.GetFieldRootPage(field.Name, _fieldsTree);
             }
         }
+
+        private void InitializeFieldRootPageForTermsVector(IndexedField field)
+        {
+            if (field.TermsVectorFieldRootPage == -1)
+            {
+                _fieldsTree ??= _transaction.CreateTree(Constants.IndexWriter.FieldsSlice);
+                _transaction.Allocator.Allocate(field.Name.Size + Constants.PhraseQuerySuffix.Length, out var memory);
```

Comment:
Can you dispose this afterward? For that matter, why not `stackalloc`?

Reply:
It's automatically cached and I don't want to accidently get AVE. It's not a big problem I think.